zarr/core.py

```diff
@@ -1238,8 +1252,12 @@ def _get_selection(self, indexer, out=None, fields=None):
             check_array_shape('out', out, out_shape)
 
         # iterate over chunks
-        if not hasattr(self.chunk_store, "getitems") or \
-           any(map(lambda x: x == 0, self.shape)):
+        if (
+            not hasattr(self.chunk_store, "getitems") and not (
+                hasattr(self.chunk_store, "get_partial_values") and
```

Comment:
Not sure about this chain of conditions. Would anything "support_efficient" without the method existing (or is it in the base class and always there?) ad why the connection to getitems?

Reply:
This if checks if we're in the code-path where we can not or should not use either `getitems` or `get_partial_values`, that's why all of those are checked. Then we use the `_chunk_getitem` code-path. Not super-happy about those code-paths, but I think cleaning those up is better left to a separate PR.